packages/dd-trace/src/config.js

```diff
@@ -285,236 +182,38 @@ class Config {
       false
     )
-    const DD_TRACE_RUNTIME_ID_ENABLED = coalesce(
-      options.experimental && options.experimental.runtimeId,
-      process.env.DD_TRACE_EXPERIMENTAL_RUNTIME_ID_ENABLED,
-      false
-    )
-    const DD_TRACE_EXPORTER = coalesce(
-      options.experimental && options.experimental.exporter,
-      process.env.DD_TRACE_EXPERIMENTAL_EXPORTER
-    )
-    const DD_TRACE_GET_RUM_DATA_ENABLED = coalesce(
-      options.experimental && options.experimental.enableGetRumData,
-      process.env.DD_TRACE_EXPERIMENTAL_GET_RUM_DATA_ENABLED,
-      false
-    )
-    const DD_TRACE_SPAN_ATTRIBUTE_SCHEMA = validateNamingVersion(
-      coalesce(
-        options.spanAttributeSchema,
-        process.env.DD_TRACE_SPAN_ATTRIBUTE_SCHEMA
-      )
-    )
-    const DD_TRACE_PEER_SERVICE_MAPPING = coalesce(
-      options.peerServiceMapping,
-      process.env.DD_TRACE_PEER_SERVICE_MAPPING ? fromEntries(
-        process.env.DD_TRACE_PEER_SERVICE_MAPPING.split(',').map(x => x.trim().split(':'))
-      ) : {}
-    )
-
-    const peerServiceSet = (
-      options.hasOwnProperty('spanComputePeerService') ||
-      process.env.hasOwnProperty('DD_TRACE_PEER_SERVICE_DEFAULTS_ENABLED')
-    )
-    const peerServiceValue = coalesce(
-      options.spanComputePeerService,
-      process.env.DD_TRACE_PEER_SERVICE_DEFAULTS_ENABLED
-    )
-
-    const DD_TRACE_PEER_SERVICE_DEFAULTS_ENABLED = (
-      DD_TRACE_SPAN_ATTRIBUTE_SCHEMA === 'v0'
-        // In v0, peer service is computed only if it is explicitly set to true
-        ? peerServiceSet && isTrue(peerServiceValue)
-        // In >v0, peer service is false only if it is explicitly set to false
-        : (peerServiceSet ? !isFalse(peerServiceValue) : true)
-    )
-
-    const DD_TRACE_REMOVE_INTEGRATION_SERVICE_NAMES_ENABLED = coalesce(
-      options.spanRemoveIntegrationFromService,
-      isTrue(process.env.DD_TRACE_REMOVE_INTEGRATION_SERVICE_NAMES_ENABLED)
-    )
-    const DD_TRACE_X_DATADOG_TAGS_MAX_LENGTH = coalesce(
-      process.env.DD_TRACE_X_DATADOG_TAGS_MAX_LENGTH,
-      '512'
-    )
-
-    const DD_TRACE_STATS_COMPUTATION_ENABLED = coalesce(
-      options.stats,
-      process.env.DD_TRACE_STATS_COMPUTATION_ENABLED,
-      isGCPFunction || isAzureFunctionConsumptionPlan
-    )
-
-    // the tracer generates 128 bit IDs by default as of v5
-    const DD_TRACE_128_BIT_TRACEID_GENERATION_ENABLED = coalesce(
-      options.traceId128BitGenerationEnabled,
-      process.env.DD_TRACE_128_BIT_TRACEID_GENERATION_ENABLED,
-      true
-    )
-
-    const DD_TRACE_128_BIT_TRACEID_LOGGING_ENABLED = coalesce(
-      options.traceId128BitLoggingEnabled,
-      process.env.DD_TRACE_128_BIT_TRACEID_LOGGING_ENABLED,
-      false
-    )
 
-    let appsec = options.appsec != null ? options.appsec : options.experimental && options.experimental.appsec
+    this.appsecOpt = options.appsec != null ? options.appsec : options.experimental && options.experimental.appsec
```

Comment:
Should this be set in `applyOptions` since it's only programmatic options that may be overridden later?

Reply:
Alternatively, it could be normalized in a copy of the options object, for example:

```js
options = {
   ...options,
  appsec: options.appsec != null ? options.appsec : options.experimental && options.experimental.appsec
}
```

Basically, the options object we store doesn't need to be the exact object that was received, it could be normalized to that using it later is easier.